ui/src/markdown.tsx

```diff
@@ -105,7 +107,18 @@ export const Markdown = ({ source }: { source: S }) => {
 
         // HACK: MarkdownIt does not support async rules.
         // https://github.com/markdown-it/markdown-it/blob/master/docs/development.md#i-need-async-rule-how-to-do-it
-        setTimeout(async () => prevHighlights.current[+codeBlockId] = await highlightSyntax(str, lang, codeBlockId), 0)
+        setTimeout(async () => prevHighlights.current[+codeBlockId] = await highlightSyntax(str, lang, codeBlockId).finally(() => {
+          // Add copy button once code block is higlighted.
+          const codeBlockContainer = document.getElementById(codeBlockId)?.parentElement
+          if (codeBlockContainer) {
+            const buttonContainer = document.createElement('span')
+            ReactDOM.render(<ClipboardCopyButton value={str} anchorElement={codeBlockContainer} showOnHoverOnly />, buttonContainer,
+              () => {
+                codeBlockContainer.style.position = 'relative'
+                codeBlockContainer.insertAdjacentElement('afterbegin', buttonContainer)
+              })
+          }
+        }), 0)
```

Comment:
* Do not mix async/await with promise callbacks.
* Why is finally needed?
* Why is ReactDOM.render callback needed?
* Can it be part of `highlightSyntax` function instead? That way, the codeBlockContainer would be available.
* Why is `insertAdjacentElement('afterbegin'` needed as opposed to an `appendChild`? I thought button is absolutely positioned so its insertion order does not matter.